galaxykit/client.py

```diff
@@ -38,22 +54,46 @@ def __init__(
         container_engine=None,
         container_registry=None,
         container_tls_verify=True,
-        https_verify=True,
+        https_verify=False,
     ):
         self.galaxy_root = galaxy_root
         self.headers = {}
         self.token = None
         self.https_verify = https_verify
-
+        
         if auth:
             if isinstance(auth, dict):
-                self.username = auth["username"]
-                self.password = auth["password"]
+                self.username = auth.get("username")
+                self.password = auth.get("password")
                 self.token = auth.get("token")
+                self.auth_url = auth.get("auth_url")
             elif isinstance(auth, tuple):
                 self.username, self.password = auth
 
-            if self.token is None:
+            token_type = "Token"
+            if self.token and self.auth_url:
+                payload = 'grant_type=refresh_token&client_id=%s&refresh_token=%s' % ('cloud-services', self.token)
+                headers = {
+                    "User-Agent": user_agent(),
+                    "Content-Type": "application/x-www-form-urlencoded",
+                }
+                resp = requests.post(
+                    self.auth_url,
+                    data=payload,
+                    verify=self.https_verify,
+                    headers=headers,
+                )
+                # import pdb;pdb.set_trace()
```

Comment:
should probably remove this stray comment.

Reply:
also do we still want all the prints below?

Reply:
👍